### PR TITLE
Applied same rules to directories, merge navbar instead of overwriting it, and allow post-_ operator to allow for generative-plugins

### PR DESCRIPTION
**Pull request by bram-nijenkamp**
- Applied same rules to directories which it did not always do.
- We now merge navbar instead of overwriting it to allow for a index.md etc as "Home" entry.
- Allow post-_ operator to allow for generative-plugins like the blog plugin which was not supported. In that case we look for an index.md and leave the rest to internal nesting.

### CODE REVIEW

**Comment by sh1nj1** [comment]
Thank you for your contribution.

> Allow post-_ operator to allow ...

Do you use ["mkdocs-blogging-plugin"](https://liang2kl.github.io/mkdocs-blogging-plugin/)?
If so, can you use existing rule for this? 

* _blog_contents
    * article1.md
* blog.md

rather than

* blog_
    * article1.md
    * index.md

File: src/mkdocs_gen_nav_plugin/plugin.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import os
-from mkdocs.plugins import BasePlugin
+from mkdocs.plugins import BasePlugin, get_plugin_logger
 from mkdocs.config import config_options
 
+log = get_plugin_logger(__name__)
```
Comment (sh1nj1) [inline src/mkdocs_gen_nav_plugin/plugin.py]: get_plugin_logger is supported from 1.5+
I would like to support 1.4.2+
So, we can use logger like this for now?
```
log = logging.getLogger('mkdocs.plugins.' + __name__)
```

Comment (bram-nijenkamp) [reply src/mkdocs_gen_nav_plugin/plugin.py]: I will change this, this week

**Comment by bram-nijenkamp** [comment]
> Thank you for your contribution.
> 
> > Allow post-_ operator to allow ...
> 
> Do you use ["mkdocs-blogging-plugin"](https://liang2kl.github.io/mkdocs-blogging-plugin/)? If so, can you use existing rule for this?
> 
>     * _blog_contents
>       
>       * article1.md
> 
>     * blog.md
> 
> 
> rather than
> 
>     * blog_
>       
>       * article1.md
>       * index.md

Not as far as I know I use this one: https://squidfunk.github.io/mkdocs-material/plugins/blog/ the reason I did it this way was because this blogging plugin doesn't seem to support naming the index file which off course conflicts with the existing behaviours that gen-nav uses. 

I did consider doing a change there but I imagine that more generative plugins use index.md as output and as such did it here.

**Comment by sh1nj1** [comment]
> Not as far as I know I use this one: https://squidfunk.github.io/mkdocs-material/plugins/blog/ the reason I did it this way was because this blogging plugin doesn't seem to support naming the index file which off course conflicts with the existing behaviours that gen-nav uses.
> 
> I did consider doing a change there but I imagine that more generative plugins use index.md as output and as such did it here.

Considering this, I found a little bit better idea.

You can use that plugin with following directory layout.
The _posts will be ignored in nav because it starts with underscore (same rule).
To include blog/index.md, gen-nav only needs ignoring index.md in the docs_dir. I am not sure why I ignored all the index.md files in paths, but only ignoring root index.md is enough and natural I think.

* blog
     * _posts
     * index.md

plugin config:
```yaml
  - blog:
      blog_dir: blog # default
      post_dir: blog/_posts
```

**Comment by bram-nijenkamp** [comment]
Ok I’ll try that tomorrow, so I make it so that only the root index is skipped. Do you want me to keep the name determining of nested index.md — using the directory name.

**Comment by sh1nj1** [comment]
I found out for the nested index.md, it needs mkdocs navigation.indexes feature

```
  features:
    - navigation.indexes
```

**Comment by sh1nj1** [comment]
@FrozenSource you can merge [this](https://github.com/FrozenSource/mkdocs-gen-nav-plugin/pull/1) first if you like, then I can merge this

**Comment by bram-nijenkamp** [comment]
Merged it, I will take a look tomorrow to confirm but your fine to merge this in I guess

**Comment by sh1nj1** [comment]
@FrozenSource Thank you - I merged and released new version

**Pull request merged**